packages/deployer/src/deploySolutionItems.ts

```diff
@@ -790,39 +791,42 @@ export function postProcessDependencies(
     // uniqueUpdateTypes must implement postProcessDependencies that should return a promise for the update
     const uniqueUpdateTypes: string[] = ["Notebook"];
 
-    const dataRequests: Array<Promise<any>> = [];
+    const modelRequests: Array<Promise<any>> = [];
     const requestedItemInfos: any = clonedSolutionsResponse.filter(
       solutionInfo => {
         if (solutionInfo.postProcess) {
-          dataRequests.push(
-            common.getItemDataAsJson(solutionInfo.id, authentication)
+          modelRequests.push(
+            deployUtils._getSolutionTemplateItem(
```

Comment:
I think this will run into issues on two fronts:
a) if the item type does not return a json response, it will throw, and
b) `_getSolutionTemplateItem` applies a schema migration, which is intended for `Solution` items

I'm more inclined to see if we can delegate all of this off to the `itemHandler` - maybe in a generic `postProcess` function. Then it can fetch what it needs, based on the item type and the actions it needs to take

Reply:
working my way through a refactor of this, I see now that a) above is irrelevant as it's already getting the /data... but we should use a more generic fn than `getSolutionTemplateItem`